testkraken/workflowregtest.py

```diff
@@ -1,154 +1,161 @@
 """Object to orchestrate worflow execution and output tests."""
 
+from copy import deepcopy
 import itertools
 import json
-import os, shutil
+import logging
+import os
+from pathlib import Path
+import shutil
 import subprocess
 import tempfile
-from copy import deepcopy
-import pdb
 
-import matplotlib
-matplotlib.use('agg')
 import pandas as pd
 import ruamel.yaml
 
 import testkraken.container_generator as cg
 import testkraken.cwl_generator as cwlg
-from testkraken.altair_plots import AltairPlots
 
 
-class WorkflowRegtest(object):
-    def __init__(self, workflow_path, base_dir=None):
-        if base_dir:
-            self.base_dir = base_dir
+class WorkflowRegtest:
+    """Object to test a workflow in many environments.
+
+    Parameters
+    ----------
+    workflow_path: Path-like, directory of workflow.
+    working_dir: Path-like, working directory, temporary directory by default.
+
+    Attributes
+    ----------
+    workflow_path: Path-like, directory of workflow.
+    working_dir: Path-like, working directory.
+    parameters: dictionary, testkraken parameters that define the workflow to
+        test, the tests to use, and the environments in which to test.
+    neurodocker_specs: dictionary, values are individual neurodocker
+        specifications (dictionaries), and keys are the SHA-1 of the
+        JSON-encoded neurodocker specifications. Tests will be run in the
+        containers that are built from these neurodocker specifications.
+    nenv: int, number of environments.
+    """
+
+    def __init__(self, workflow_path, working_dir=None):
+        self.workflow_path = Path(workflow_path).absolute()
+        if working_dir is None:
+            self.working_dir = Path(tempfile.mkdtemp(
```

Comment:
I would prefer to keep previous structure of directory as default (and not temporary). At the end the idea is to  collect results, dashboard etc.
we can have a flag as an additional argument